services/QuillLMS/engines/evidence/app/models/evidence/activity.rb

```diff
@@ -18,6 +18,7 @@ class Activity < ApplicationRecord
 
     before_destroy :expire_turking_rounds
     before_validation :set_parent_activity, on: :create
+    before_validation :increment_version, on: :update
```

Comment:
Is there any worry around this causing version changes when it's not a meaningful change?  Like, tweaking the text of the title or changing the value in the `notes` attribute?  I can imagine a scenario where Curriculum changes the ML model (and we want a version update), then a week later tweaks the title, and suddenly we have a discontinuity where the grading data before and after the title change should be grouped together, but aren't.

It's possible that the rest of the code that will come after this migration addresses the issue, in which case you can ignore this concern, but I wanted to raise it just in case.

Reply:
I'd like to hear more about this too.

Reply:
Good point. I removed this logic from the `update` callback. We will increment via a single UX and API endpoint, so every version bump occurs when a human decides.